Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/interceptor/HeaderRequestOutInterceptor.java

```diff
@@ -0,0 +1,81 @@
+ *       and/or other materials provided with the distribution.
+ *     * Neither the name of the United States Government nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE UNITED STATES GOVERNMENT BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.messaging.client.interceptor;
+
+import org.apache.cxf.transports.http.configuration.ConnectionType;
+
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.cxf.interceptor.Fault;
+import org.apache.cxf.message.Message;
+import org.apache.cxf.phase.AbstractPhaseInterceptor;
+import org.apache.cxf.phase.Phase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author mpnguyen
+ *
+ */
+public class HeaderRequestOutInterceptor extends AbstractPhaseInterceptor<Message> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HeaderRequestOutInterceptor.class);
+
+    public HeaderRequestOutInterceptor() {
+        super(Phase.PRE_LOGICAL);
+    }
+
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.apache.cxf.interceptor.Interceptor#handleMessage(org.apache.cxf.message.Message)
+     */
+    @Override
+    public void handleMessage(final Message message) throws Fault {
+        try {
+            Map<String, List> headers = (Map<String, List>) message.get(Message.PROTOCOL_HEADERS);
+            LOG.debug("HeaderRequestOutInterceptor");
+            if (headers == null) {
+                headers = new HashMap<String, List>();
+            }
+            final Object keepAliveDuration = message.get(NhincConstants.KEEPALIVE_DURATION);
+            LOG.debug("Keep Alive duration {} seconds", keepAliveDuration);
+            if (keepAliveDuration != null) {
+                headers.put(ConnectionType.KEEP_ALIVE.value(),
+                        Collections.singletonList("timeout=" + keepAliveDuration));
+                message.put(Message.PROTOCOL_HEADERS, headers);
+            }
+        } catch (final Exception ce) {
```

Comment:
Is Exception the actual type that needs to be caught, or can this be replaced with a multicatch?